src/pybind/mgr/telemetry/module.py

```diff
@@ -982,6 +1211,73 @@ def send(self,
+        # Formatting the perf histograms so they are human-readable. This will change the
+        # ranges and values, which are currently in list form, into strings so that
+        # they are displayed horizontally instead of vertically.
+        try:
+            # Formatting ranges and values in osd_perf_histograms
+            modes_to_be_formatted = ['osd_perf_histograms_aggregated', 'osd_perf_histograms_separated']
+            for mode in modes_to_be_formatted:
+                for config in report[mode]:
+                    for histogram in config:
+                        # Adjust ranges by converting lists into strings
+                        for axis in config[histogram]['axes']:
+                            for i in range(0, len(axis['ranges'])):
+                                axis['ranges'][i] = str(axis['ranges'][i])
+                        # Adjust values by converting lists into strings
+                        if mode == 'osd_perf_histograms_aggregated':
+                            for i in range(0, len(config[histogram]['values'])):
+                                config[histogram]['values'][i] = str(config[histogram]['values'][i])
+                        else: # if mode == 'osd_perf_histograms_separated'
+                            for osd in config[histogram]['osds']:
+                                for i in range(0, len(osd['values'])):
+                                    osd['values'][i] = str(osd['values'][i])
+        except KeyError:
+            # If the perf channel is not enabled, there should be a KeyError since
+            # 'osd_perf_histograms' would not be present in the report. In that case,
+            # the show function should pass as usual without trying to format the
+            # histograms.
+            pass
+
+    def toggle_channel(self, action: str, channels: Optional[List[str]] = None) -> Tuple[int, str, str]:
+        '''
+        Enable or disable a list of channels
+        '''
+        if not self.enabled:
+            # telemetry should be on for channels to be toggled
+            msg = 'Telemetry is off. Please consider opting-in with `ceph telemetry on`.\n' \
+                  'Preview sample reports with `ceph telemetry preview`.'
+            return 0, msg, ''
+
+        if channels is None:
+            msg = f'Please provide a channel name. Available channels: {ALL_CHANNELS}.'
+            return 0, msg, ''
+
+        state = action == 'enable'
+        msg = ''
+        for c in channels:
+            if c not in ALL_CHANNELS:
+                msg = f"{msg}{c} is not a valid channel name. "\
+                        f"Available channels: {ALL_CHANNELS}.\n"
+            else:
+                self.set_module_option(f"channel_{c}", state)
+                setattr(self,
+                        f"channel_{c}",
+                        state)
+                msg = f"{msg}channel_{c} is {action}d\n"
+
+        return 0, msg, ''
```

Comment:
mgr_module gives you a `HandleCommandResult` helper:
```suggestion
        return HandleCommandResult(stdout=msg) # default is retval = 0 and empty stderr
```